esrally/mechanic/supplier.py

```diff
@@ -694,6 +757,155 @@ def run(self, command, override_src_dir=None):
+        """
+        Given a JDK major release version, find a suitable docker image to use for compiling ES & ES plugins from sources
+        :param build_jdk: The build JDK major version required to build Elasticsearch and plugins.
+        :return: The Docker image name and respective tag that satisfies the 'build_jdk' requirement.
+        """
+        logger = logging.getLogger(__name__)
+        # Temurin only maintains images for JDK 11, 17, 18 (at time of writing)
+        # older builds of ES may require specifics like JDK 14 etc, in which case we rely on the OpenJDK images
+        #
+        # We modify the base image (create user/group, install git), so we need to ensure the conatiner's base image
+        # is Debian based (apt required)
+        if build_jdk >= 17:
+            docker_image_name = f"eclipse-temurin:{build_jdk}-jdk-jammy"
+            logger.info("Build JDK version [%s] is >= 17, using Docker image [%s].", build_jdk, docker_image_name)
+        # There is no Debian based OpenJDK JDK 12 image
+        elif build_jdk == 12:
+            docker_image_name = "adoptopenjdk/openjdk12:debian"
+        else:
+            docker_image_name = f"openjdk:{build_jdk}-jdk-buster"
+            logger.info("Build JDK version [%s] is < 17, using Docker image [%s].", build_jdk, docker_image_name)
+        return docker_image_name
+
+    def stop_containers(self, container_names):
+        for c in container_names:
+            try:
+                existing_container = self.client.containers.get(c)
+                existing_container.stop()
+                existing_container.remove()
+            except docker.errors.NotFound:
+                pass
+            except docker.errors.APIError as e:
+                raise exceptions.SystemSetupError(self.err_msg(e))
+
+    def tail_container_logs(self, output, container_name):
+        try:
+            with open(self.log_file, "a+", encoding="utf-8") as f:
+                while True:
+                    line = next(output).decode("utf-8")
+                    f.write(line)
+                    f.flush()
+        except StopIteration:
+            self.logger.info("Log stream ended for [%s]", container_name)
+
+    def check_container_return_code(self, completion, container_name):
+        if completion["StatusCode"] != 0:
+            raise SystemSetupError(
+                f"Docker container [{container_name}] failed with status code [{completion['StatusCode']}]: Error [{completion['Error']}]"
+            )
+        self.logger.info("Container [%s] completed successfully.", container_name)
+
+    def create_base_container_image(self):
+        try:
+            # create a new image with the required users & git installed
+            image_container = self.client.containers.run(
+                detach=True,
+                name=self.image_builder_container_name,
+                image=self.resolve_jdk_build_container_image(self.build_jdk),
```

Comment:
maybe we specify the new image name somewhere? atm it's nameless (and it helps troubleshooting)